### PR TITLE
Mediawiki - Several Improvements

### PR DESCRIPTION
**Pull request by juan131**
**What this PR does / why we need it**:

This PR provides several improvements to Mediawiki Helm Chart:

 - Add Ingress and TLS Secret to allow the user configure a Ingress Rule for Mediawiki
 - Improve configurable parameters for services, readiness/liveness probes, etc.
 - Improve README.md file and helpers
 - Update MariaDB requirement.

### CODE REVIEW

File: stable/mediawiki/requirements.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 dependencies:
 - name: mariadb
-  version: 0.7.0
+  version: 2.1.18
```
Comment (carrodher) [inline stable/mediawiki/requirements.yaml]: Can you use `2.x.x` instead of `2.1.18`?

File: stable/mediawiki/README.md [hunk 2]
```diff
@@ -45,41 +45,66 @@ The command removes all the Kubernetes components associated with the chart and
...
-| `resources`                          | CPU/Memory resource requests/limits      | Memory: `512Mi`, CPU: `300m`                            |
+|              Parameter               |               Description                                   |                         Default                         |
+|--------------------------------------|-------------------------------------------------------------|---------------------------------------------------------|
+| `image.registry`                     | MediaWiki image registry                                    | `docker.io`                                             |
+| `image.repository`                   | MwdiaWiki Image name                                        | `bitnami/mediawiki`                                     |
+| `image.tag`                          | MediaWiki Image tag                                         | `{VERSION}`                                             |
+| `image.pullPolicy`                   | Image pull policy                                           | `Always` if `imageTag` is `latest`, else `IfNotPresent` |
```
Comment (tompizmor) [inline stable/mediawiki/README.md]: Change the default value to `Always`

File: stable/mediawiki/templates/_helpers.tpl [hunk 3]
```diff
@@ -15,6 +15,20 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
+*/}}
+{{- define "mediawiki.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Return the proper Mediawiki image name
+*/}}
+{{- define "mediawiki.image" -}}
+{{- printf "%s/%s:%s" .Values.image.registry .Values.image.repository .Values.image.tag -}}
```
Comment (tompizmor) [inline stable/mediawiki/templates/_helpers.tpl]: check with @carrodher as he is fixing a corner case with this image functions

File: stable/mediawiki/templates/apache-pvc.yaml [hunk 4]
```diff
@@ -5,9 +5,9 @@ metadata:
   name: {{ template "mediawiki.fullname" . }}-apache
   labels:
     app: {{ template "mediawiki.fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+    chart: {{ template "mediawiki.chart" . }}
+    release: {{ $.Release.Name | quote }}
```
Comment (tompizmor) [inline stable/mediawiki/templates/apache-pvc.yaml]: is the `$` necessary?

File: stable/mediawiki/templates/deployment.yaml [hunk 5]
```diff
@@ -95,12 +105,12 @@ spec:
           httpGet:
             path: /index.php
             port: http
-          initialDelaySeconds: 120
+{{ toYaml .Values.livenessProbe | indent 10 }}
         readinessProbe:
```
Comment (tompizmor) [inline stable/mediawiki/templates/deployment.yaml]: it should also be possible to completely disable the liveness/readiness probes

File: stable/mediawiki/templates/tls-secrets.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.ingress.enabled }}
+{{- range .Values.ingress.secrets }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .name }}
```
Comment (tompizmor) [inline stable/mediawiki/templates/tls-secrets.yaml:6]: is this name ok?

**Review by tompizmor — CHANGES_REQUESTED** [review]

File: stable/mediawiki/requirements.yaml (hunk 1 above)
Comment (juan131) [reply stable/mediawiki/requirements.yaml]: 👍

File: stable/mediawiki/templates/_helpers.tpl (hunk 3 above)
Comment (juan131) [reply stable/mediawiki/templates/_helpers.tpl]: @carrodher do you see anything weird?

File: stable/mediawiki/templates/apache-pvc.yaml (hunk 4 above)
Comment (juan131) [reply stable/mediawiki/templates/apache-pvc.yaml]: Nope

File: stable/mediawiki/templates/tls-secrets.yaml (hunk 6 above)
Comment (juan131) [reply stable/mediawiki/templates/tls-secrets.yaml:6]: I think so

File: stable/mediawiki/templates/deployment.yaml (hunk 5 above)
Comment (juan131) [reply stable/mediawiki/templates/deployment.yaml]: 👍

File: stable/mediawiki/templates/tls-secrets.yaml (hunk 6 above)
Comment (tompizmor) [reply stable/mediawiki/templates/tls-secrets.yaml:6]: yes, it is fine. I didn't realize it was inside a `range`

File: stable/mediawiki/templates/_helpers.tpl (hunk 3 above)
Comment (tompizmor) [reply stable/mediawiki/templates/_helpers.tpl]: maybe you should define like we do in ES and other charts, to be consistent:
https://github.com/bitnami/charts/blob/master/bitnami/elasticsearch/templates/_helpers.tpl#L25-L32

Comment (juan131) [reply stable/mediawiki/templates/_helpers.tpl]: I don't want to use "defaults". Defaults should be provided by "values.yaml"

File: stable/mediawiki/templates/tls-secrets.yaml (hunk 6 above)
Comment (juan131) [reply stable/mediawiki/templates/tls-secrets.yaml:6]: 👍

**Review by tompizmor — APPROVED** [review]

**Pull request merged**